core/block/source/source.go

```diff
@@ -364,12 +370,13 @@ func (s *source) buildChange(params PushChangeParams) (c *pb.Change) {
 	if params.DoSnapshot || s.needSnapshot() || len(params.Changes) == 0 {
 		c.Snapshot = &pb.ChangeSnapshot{
 			Data: &model.SmartBlockSnapshotBase{
-				Blocks:        params.State.BlocksToSave(),
-				Details:       params.State.Details(),
-				ObjectTypes:   domain.MarshalTypeKeys(params.State.ObjectTypeKeys()),
-				Collections:   params.State.Store(),
-				RelationLinks: params.State.PickRelationLinks(),
-				Key:           params.State.UniqueKeyInternal(),
+				Blocks:                   params.State.BlocksToSave(),
+				Details:                  params.State.Details(),
+				ObjectTypes:              domain.MarshalTypeKeys(params.State.ObjectTypeKeys()),
+				Collections:              params.State.Store(),
+				RelationLinks:            params.State.PickRelationLinks(),
+				Key:                      params.State.UniqueKeyInternal(),
+				OriginalCreatedTimestamp: params.State.OriginalCreatedTimestamp(),
```

Comment:
we have a bunch of places where snapshot is filled (export for example), maybe add this there too?

Reply:
When we do import we use the createdDate from the details instead. Because it may be both "original date" and "object tree date". Also this way it backward compatible and less logic to handle. At the same time these 2 places may be ambiguous if someone want to operate with snapshots models.

Maybe we can have it unexported?

Reply:
Ah, we can't make it unexported here.. Only via creating separate message struct with omitted field only for the export/import, which is too much for this.
I've added the comment to the models.proto